src/sirocco/parsing/_yaml_data_models.py

```diff
@@ -25,46 +26,67 @@
 
 
 class _NamedBaseModel(BaseModel):
-    """Base class for all classes with a key that specifies their name.
-
-    For example:
+    """
+    Base model for reading names from yaml keys *or* attributes.
 
-    .. yaml
+    Reading from attributes is allowed in order to enable the standard
+    constructor usage from Python, as demonstrated in the below examples.
```

Comment:
I must say this is quite pedantic but I had some troubles to understand the text: If I understand the [yaml specification](https://yaml.org/spec/1.2.2/#nodes), they dont use the termology "attributes" but "values"

```suggestion
    Reading from values is allowed in order to enable the standard
    constructor usage from Python, as demonstrated in the below examples.
```